File: leetecode/LargestNumber.java
import java.util.Arrays;
import java.util.Stack;

class Solution {
    public String largestNumber(int[] nums) {
        Integer[] input = Arrays.stream(nums).boxed().toArray(Integer[]::new);
        Arrays.sort(input,
                   (a,b)->{
                       String s1 = a.toString();
                       String s2 = b.toString();
                       return (s2+s1).compareTo(s1+s2);
                   }
                   );
        StringBuilder sb = new StringBuilder();
        for(int x:input){
            sb.append(String.valueOf(x));
        }
        return sb.toString();
    }
    public static void main(String[] args) {
        Solution s = new Solution();
        int[] input = {34323,3432};
        System.out.println(s.largestNumber(input));
    }
}
//[34323,3432]

File: leetecode/LargestNumber.py
from functools import cmp_to_key
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        if(nums.count(0)==len(nums)): return "0"
        def cmp(a,b):
            aa = str(a)
            bb = str(b)
            f = aa+bb
            g = bb+aa
            if(f<g): return 1
            elif(f>g): return -1
            else: return 0
        nums = sorted(nums,key=cmp_to_key(cmp))
        res = "".join([str(x) for x in nums]);
        return res